torch/csrc/jit/frontend/ir_emitter.cpp

```diff
@@ -1733,6 +1737,83 @@ struct to_ir {
     emitLoopCommon(stmt.range(), emit_body, nullptr, {}, cond);
   }
 
+  void emitWith(const With& stmt) {
+    auto targets = stmt.targets();
+    // Keep a stack of entered objects so they can be exited
+    // in the right order.
+    std::stack<Value*> entered;
+
+    for (const auto& target : targets) {
+      Expr e = target.target();
+
+      auto* rhs = emitExpr(e);
+      auto* n = graph->insertNode(graph->create(prim::Enter, {rhs}));
+      entered.push(rhs);
+      auto rhsClass = rhs->type()->expect<ClassType>();
+
+      if (!rhsClass) {
+        throw ErrorReport(e.range())
+            << "With item expression does not return a class type";
+      }
+
+      auto* enterMethod = rhsClass->getMethod("__enter__");
+      auto* exitMethod = rhsClass->getMethod("__exit__");
+
+      if (!enterMethod || !exitMethod) {
+        throw ErrorReport(e.range())
+            << "Object returned by with item expression does not define __enter__ and __exit__ methods";
+      }
+
+      // Check the schema of __enter__.
+      auto& enterSchema = enterMethod->getSchema();
+      if (enterSchema.arguments().size() != 1 ||
+          enterSchema.returns().size() != 1) {
+        throw ErrorReport(e.range())
+            << "__enter__ must have only one argument and one return value";
+      }
+
+      // Check the schema of __exit__.
+      auto& exitSchema = exitMethod->getSchema();
+      if (exitSchema.arguments().size() != 4 ||
+          exitSchema.returns().size() != 1) {
+        throw ErrorReport(e.range())
+            << "__exit__ must have four arguments and no return value";
+      } else {
+        if (exitSchema.returns().at(0).type() != NoneType::get()) {
+          throw ErrorReport(e.range()) << "__exit__ must have no return value";
+        }
+
+        for (unsigned i = 1; i < 4; ++i) {
+          if (exitSchema.arguments().at(i).type() != AnyType::get()) {
+            throw ErrorReport(e.range())
+                << "argument " << i << " of __exit__ must have Any type";
```

Comment:
can you add some context to this error message ? it won't clear to a user why we require `Any` type so pretty much everyone will hit this. Maybe something like:

```argument i of __exit__ must have Any type. TorchScript does not currently support passing type, value, or traceback to the __exit__ function.```

or something else as you see fit.